Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1909 @@
+        s_media_buffer_32[2] = 1;
+        s_media_buffer_32[4] = 0xFF;
+        break;
+      case AMMBCommand::GetSystemFlags:
+        s_media_buffer[4] = 1;
+        s_media_buffer[5] = GDROM;
+        // Enable development mode (Sega Boot)
+        // This also allows region free booting
+        s_media_buffer[6] = 1;
+        s_media_buffer[8] = 0;  // Access Count
+        break;
+      case AMMBCommand::GetMediaBoardSerial:
+        memcpy(s_media_buffer + 4, "A89E-27A50364511", 16);
+        break;
+      case AMMBCommand::Unknown_104:
+        s_media_buffer[4] = 1;
+        break;
+      case AMMBCommand::NetworkReInit:
+        break;
+      case AMMBCommand::TestHardware:
+        // Test type
+
+        // 0x01: Media board
+        // 0x04: Network
+
+        DEBUG_LOG_FMT(AMMEDIABOARD, "GC-AM: TestHardware: ({:08x})", s_media_buffer_32[11]);
+        // Pointer to a s_memory address that is directly displayed on screen as a string
+        DEBUG_LOG_FMT(AMMEDIABOARD, "GC-AM:               ({:08x})", s_media_buffer_32[12]);
+
+        // On real s_system it shows the status about the DIMM/GD-ROM here
+        // We just show "TEST OK"
+        s_memory.Write_U32(0x54534554, s_media_buffer_32[12]);
+        s_memory.Write_U32(0x004B4F20, s_media_buffer_32[12] + 4);
+
+        s_media_buffer_32[1] = s_media_buffer_32[9];
+        break;
+      case AMMBCommand::Closesocket:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[10])];
+
+        const int ret = closesocket(fd);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: closesocket( {}({}) ):{}\n", fd,
+                       s_media_buffer_32[10], ret);
+
+        s_sockets[SocketCheck(s_media_buffer_32[10])] = SOCKET_ERROR;
+
+        s_media_buffer_32[1] = ret;
+        s_last_error = SSC_SUCCESS;
+      }
+      break;
+      case AMMBCommand::Connect:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[10])];
+        const u32 off = s_media_buffer_32[11] - NetworkCommandAddress1;
+        const u32 len = s_media_buffer_32[12];
+
+        if (NetworkCMDBufferCheck(off + sizeof(sockaddr_in)))
+        {
+          break;
+        }
+
+        sockaddr_in addr;
+        memcpy(&addr, s_network_command_buffer + off, sizeof(sockaddr_in));
+
+        const int ret = NetDIMMConnect(fd, &addr, len);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: connect( {}({}), ({},{}:{}), {} ):{}\n", fd,
+                       s_media_buffer_32[10], addr.sin_family, inet_ntoa(addr.sin_addr),
+                       Common::swap16(addr.sin_port), len, ret);
+
+        s_media_buffer[1] = s_media_buffer[8];
+        s_media_buffer_32[1] = ret;
+      }
+      break;
+      case AMMBCommand::Recv:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[10])];
+        const u32 off = s_media_buffer_32[11];
+        const auto len = std::min<u32>(s_media_buffer_32[12], sizeof(s_network_buffer));
```

Comment:
This needs to take into account that the maximum allowed length is smaller if `off` is larger than `NetworkBufferAddress5`.